### ISSUE TITLE
hab pkg binlink attempts to binlink build dependencies

### ISSUE THREAD

**Issue by adamleff** [habitat-sh/habitat#3172]
Hello!

It appears `hab pkg binlink` attempts to binlink all dependencies, not just runtime dependencies. When running `hab pkg binlink chef/inspec`, it errors out with:

```
» Symlinking ri from chef/inspec into /hab/bin
★ Binary ri from chef/inspec/1.37.6/20170914182855 symlinked to /hab/bin/ri
» Symlinking rake from chef/inspec into /hab/bin
★ Binary rake from chef/inspec/1.37.6/20170914182855 symlinked to /hab/bin/rake
✗✗✗
✗✗✗ No such file or directory (os error 2)
✗✗✗
```

Running with `RUST_LOG=debug RUST_BACKTRACE=1` does not yield any additional information prior to the error.

I `strace`d the process and found it errored out on this `open` call:

```
open("/hab/pkgs/core/bundler/1.15.4/20170830210853/bin", O_RDONLY|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
```

This is a valid error because that path does not exist because `core/bundler` is a build dependency for `chef/inspec`, not a runtime dependency:

https://github.com/chef/inspec/blob/master/habitat/plan.sh#L19

... so it doesn't exist on a machine / studio where `chef/inspec` was installed but not necessarily built.

- [x] The OS (including version) where you are running any of the Habitat commands.

macOS 10.12.6, Docker for Mac 17.06.2-ce-mac27

- [x] Debug/backtrace of the command you are trying to run. You can set the following environment variables before running the `hab` command to generate a trace:

```
[8][default:/src:1]# RUST_LOG=debug hab pkg binlink chef/inspec
DEBUG:habitat_common::ui: InputStream(stdin): { is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stdout): { is_colored(): true, supports_color(): true, is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stderr): { is_colored(): true, supports_color(): true, is_a_terminal(): true }
DEBUG:hab: clap cli args: ["hab", "pkg", "binlink", "chef/inspec"]
DEBUG:hab: remaining cli args: []
» Symlinking inspec from chef/inspec into /hab/bin
★ Binary inspec from chef/inspec/1.37.6/20170914182855 symlinked to /hab/bin/inspec
... etc - no additional info provided around the error
```
		
- [x] Current Habitat environment variables where the `hab` command or supervisor is running.  

	- [x] If you're running into problems with the [Habitat Studio](https://www.habitat.sh/docs/concepts-studio/), please use the `env | grep ...` command specified above to include environment variables:
		- [x] a. before starting the studio

None.

		- [x] b. inside the studio

```
[1][default:/src:0]# env | grep HAB | grep -v HAB_AUTH_TOKEN
HAB_ORIGIN=adamleff
HAB_BINLINK_DIR=/hab/bin
```

**Comment by rsertelon** [comment]
@adamleff Is this still a problem for you? I've just tested the following and no errors on my side:

```
hab studio enter
hab pkg install chef/inspec
hab pkg binlink chef/inspec
```
Output for binlink:
```
[3][default:/src:0]# hab pkg binlink chef/inspec
» Symlinking inspec from chef/inspec into /hab/bin
★ Symlinked inspec from chef/inspec/1.43.8/20171102203700 to /hab/bin/inspec
```

**Comment by adamleff** [comment]
@rsertelon yeah, looks fixed now! Re: #2376, it should work if you binlink in `core/ruby` as well. That fixed it for me. I suppose there's a larger discussion as to whether or not DEPS should have their stuff binlink'd automatically.
